### PR TITLE
(fix) Workspaces not maximizing; add workspace animations

### PR DESCRIPTION
**Pull request by brandones**
# Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. Ensure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing#contributing-guidelines) label (such as `feat`, `fix`, or `chore`, among others). See existing PR titles for inspiration.

## For changes to apps

- [x] My work conforms to the [**O3 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://om.rs/o3ui).

## If applicable

- [ ] My work includes tests or is validated by existing tests.
- [ ] I have updated the [esm-framework mock](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/mock.tsx) to reflect any API changes I have made.

## Summary

Fixes the bug documented in the ticket. Also adds animations per the designs!

## Screenshots

https://github.com/openmrs/openmrs-esm-core/assets/1031876/ef479b59-ae40-4a34-95eb-d5d985ada388

https://github.com/openmrs/openmrs-esm-core/assets/1031876/6658839c-8406-4233-9bef-efd7d3006266

## Related Issue
https://openmrs.atlassian.net/browse/O3-3501

### CODE REVIEW

File: packages/framework/esm-styleguide/src/workspaces/container/workspace-window.component.tsx [hunk 1]
```diff
@@ -76,82 +81,90 @@ function Workspace({ workspaceInstance, additionalWorkspaceProps }: WorkspacePro
         styles.workspaceWindowSpacer,
         width === 'narrow' ? styles.narrowWorkspace : styles.widerWorkspace,
         {
-          [styles.maximizedWindow]: isMaximized,
-          [styles.hidden]: isHidden,
+          [styles.hiddenRelative]: isHidden,
         },
       )}
     >
-      <div className={styles.workspaceFixedContainer}>
```
Comment (brandones) [inline packages/framework/esm-styleguide/src/workspaces/container/workspace-window.component.tsx:84]: Use "hide whitespace" to review

**Review by denniskigen — APPROVED** [review]
LGTM! Thanks, @brandones!

**Comment by ojwanganto** [comment]
Thanks, @brandones

**Comment by donaldkibet** [comment]
@brandones, thank you for addressing this. However, it seems that this commit has introduced a regression where workspace items lose their state when switching between workspace types (e.g., clinical forms lose changes once a different workspace is opened).

regressionWorkspace

**Comment by brandones** [comment]
Thanks for the report @donaldkibet . I [requested](https://openmrs.atlassian.net/browse/O3-3361) an E2E test for this a while ago; hopefully we can get that soon to prevent this kind of thing.

Filed https://openmrs.atlassian.net/browse/O3-3520

**Pull request merged**